src/components/common/App/AppSelect.vue

```diff
@@ -4,25 +4,31 @@ export default defineComponent({
   props: {
     options: {
       type: Array,
-    }
+    },
+    optionStart: {
+      type: String,
+      required: true,
+    },
   },
   data() {
     return {
-      optionSelected: 'All movies',
-      optionDefault: 'All movies'
+      optionSelected: this.optionStart,
```

Comment:
So while all of this works, we actually can do better than having both a prop for some data and also local data entry for `optionSelected`.

In cases of custom components which deal with data it's a good practice to implement a custom `v-model`. Not only does this make it much more clear about the intent (and gives us appropriate events to handle some peculiar cases) but also limits the amount of bugs we encounter.

https://vuejs.org/guide/components/events.html#usage-with-v-model

In general all you need is a small tweak here - we need to accept a prop which will hold initial/current value and emit appropriate event when the value changes. The names of these are in some way up to you - there are defaults (more on that in the doc above) but you can change that.

Then tweak our template a little bit so that we bind the select's `value` prop and handle it's `change` event.

```vue
<select :value="yourValueProp" @change="handleValueChange">
```